File: DataStructure/leetcode974/sloveLeetcode974.cpp
// class Solution 
// {
// public:
//     int subarraysDivByK(vector<int>& nums, int k) 
//     {
//         unordered_map<int,int>hash={{0,1}};

//         const int n=nums.size();
//         int i,sum=0,modulus,ans=0;
//         for(i=0;i<n;++i)
//         {
//             sum+=nums[i];
//             modulus=(sum%k + k)%k;

//             //count return 1:key exist return 0:key non-exist
//             if(hash.count(modulus))
//                 ans+=hash[modulus];
//             ++hash[modulus];
//         }    

//         return ans;
//     }
// };

File: DataStructure/prefix/leetcode974/sloveLeetcode974.go
// func subarraysDivByK(nums []int, k int) int {
//     hash:=make(map[int]int)
//     hash[0]=1

//     sum:=0
//     var module int
//     ans:=0
//     for _,num:=range nums{
//         sum+=num

//         module=(sum%k+k)%k

//         _,exsit:=hash[module]
//         if exsit{
//             ans+=hash[module]
//         }
//         hash[module]++
//     }

//     return ans;
// }